python/spark_sklearn/grid_search.py

```diff
@@ -180,35 +240,40 @@ def fit(self, X, y=None):
         """
         return self._fit(X, y, ParameterGrid(self.param_grid))
 
-    def _fit(self, X, y, parameter_iterable):
-        """Actual fitting,  performing the search over parameters."""
+    
+    def fit(self, X, y=None, groups=None, **fit_params):
```

Comment:
In sklearn 0.18 they got rid of _fit() and just used fit() (see: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_search.py). I changed it to be consistent with them.